cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/cube/DefaultCube.java

```diff
@@ -130,6 +130,31 @@ SELECT count('read.ops')                                     << measure name and
     return convertToQueryResult(query, resultTable);
   }
 
+  /**
+   * Deletes the data specified by {@link CubeQuery} from all the fact tables.
+   * @param query query specifies parameters for deletion. groupByTags and resolution fields are not used.
+   * @throws Exception
+   */
+  @Override
+  public void delete(CubeQuery query) throws Exception {
+    List<TagValue> tagValues = Lists.newArrayList();
+
+    // find all the aggregations that match the sliceByTags in the query and
+    // use the tag values of the aggregation to delete entries in all the fact-tables.
```

Comment:
you can add a comment that this may be very inefficient, so better use TTL instead of this method. This one is mainly to support existing old functionality